### PR TITLE
Document support for SameSite cookie directive

### PR DESCRIPTION
**Pull request by eleftherias**
Resolves: #1517

### CODE REVIEW

**Comment by eleftherias** [comment]
@vpavic Sounds good. 
I will add an entry in the API Documentation section about `DefaultCookieSerialzer`, is that what you had in mind?

**Comment by vpavic** [comment]
Yes, that's what I was aiming for. Thanks.

**Comment by eleftherias** [comment]
@vpavic I have added the new section to the API Documentation. Thanks for your feedback and let me know any further thoughts.

File: spring-session-docs/src/docs/asciidoc/index.adoc [hunk 1]
```diff
@@ -1230,6 +1231,67 @@ Note that if you use Hazelcast's `MapStore` to persist your sessions `IMap`, the
...
+With some implementations (that is, Redis) this option provides no performance benefit.
+However, it can help with tracing logs of a particular user.
+* `domainName`: Allows specifying a specific domain name to be used for the cookie.
+This option is simple to understand but often requires a different configuration between development and production environments.
+See `domainNamePattern` as an alternative.
+* `domainNamePattern`: A case-insensitive pattern used to extract the domain name from the `HttpServletRequest#getServerName()`.
+The pattern should provide a single grouping that is used to extract the value of the cookie domain.
+If the regular expression does not match, no domain is set and the existing domain is used.
+If the regular expression matches, the first https://docs.oracle.com/javase/tutorial/essential/regex/groups.html[grouping] is used as the domain.
+* `sameSite`: The value for the `SameSite` cookie directive.
```
Comment (vpavic) [inline spring-session-docs/src/docs/asciidoc/index.adoc]: I think we should mention here that `null` could be used as value to disable the serialization of `SameSite` directive.

Comment (eleftherias) [reply spring-session-docs/src/docs/asciidoc/index.adoc]: Done

**Pull request closed without merging**